### PR TITLE
Issue/#176 default user value

### PR DESCRIPTION
**Pull request by abdulmansour**
# Related Issue
- [link to the issue on zenhub](https://app.zenhub.com/workspaces/moneytree-board-5f5bcf1c7ea48f001c28c973/issues/dpigeon/money-tree/176)

# Proposed changes
- newly created user will have a predefined avatarURL pointing to a default image stored in s3
- when the user updates his default image, the default image is NOT deleted from s3
- endpoint exclusively used for updating image, change to => PUT /users/profile-picture/{id}

# Reviewer(s)
 - @razine-bensari @DPigeon @Alessandro100

### CODE REVIEW

**Review by Alessandro100 — CHANGES_REQUESTED** [review]
This PR is going from your branch into master, please change it so it goes into the dev branch instead, after that this feature can be properly reviewed (right now there are 143 files changed)

File: backend/src/main/resources/application-local.properties [hunk 1]
```diff
@@ -25,4 +25,5 @@ IEXCloud.secret.token=${IEXCLOUD_SECRET_TOKEN_SANDBOX}
 # Amazon S3 related properties
 aws.access.key.id=${AWS_ACCESS_KEY}
 aws.secret.access.key=${AWS_SECRET_ACCESS_KEY}
-aws.profile.pictures.bucket=${AWS_PROFILE_PICTURES_BUCKET}
\ No newline at end of file
+aws.profile.pictures.bucket=${AWS_PROFILE_PICTURES_BUCKET}
+aws.default.profile.picture=${DEFAULT_PROFILE_PICTURE}
```
Comment (razine-bensari) [inline backend/src/main/resources/application-local.properties:29]: You do not need to store it as env variable, way too much work. The url does not need to be protected.
having `aws.default.profile.picture=https://my.default.url` is good enough

File: backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java [hunk 2]
```diff
@@ -37,14 +37,16 @@
    private final AmazonS3Service amazonS3Service;
 
    private final String bucketName;
+   private final String defaultProfileURL;
 
    @Autowired
-   public DefaultUserService(UserDao userDao, ValidatorFactory validatorFactory, AmazonS3Service amazonS3Service, @Value("${aws.profile.pictures.bucket}") String bucketName) {
+   public DefaultUserService(UserDao userDao, ValidatorFactory validatorFactory, AmazonS3Service amazonS3Service, @Value("${aws.profile.pictures.bucket}") String bucketName, @Value("${aws.default.profile.picture}") String defaultProfileURL) {
       this.userDao = userDao;
```
Comment (razine-bensari) [inline backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java:44]: It does not make sense to pass a default url as constructor. You can inject it within the field attribute.

File: backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java [hunk 3]
```diff
@@ -37,14 +37,16 @@
    private final AmazonS3Service amazonS3Service;
 
    private final String bucketName;
+   private final String defaultProfileURL;
```
Comment (razine-bensari) [inline backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java:40]:
```
@Value("${aws.profile.pictures.bucket}")
private final String defaultProfileURL;
```

**Review by razine-bensari — CHANGES_REQUESTED** [review]
Please update the default service constructor.

Also I think there will be some conflicts if David's brnach is merged first regarding generic user endpoint

File: backend/src/main/resources/application-local.properties (hunk 1 above)
Comment (abdulmansour) [reply backend/src/main/resources/application-local.properties:29]: sounds good

File: backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java (hunk 2 above)
Comment (abdulmansour) [reply backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java:44]: right agreed, was coded at 2am 🤦‍♂️

File: backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java (hunk 3 above)
Comment (abdulmansour) [reply backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java:40]: this relates to the previous comment, will make the adjustment

**Review by razine-bensari — APPROVED** [review]
Nice work!, Just make sure it builds green before merging

**Review by Alessandro100 — APPROVED** [review]
nice changes

**Pull request merged**